src/@types/react-app-env.d.ts

```diff
@@ -0,0 +1,21 @@
+/* eslint-disable @typescript-eslint/triple-slash-reference */
+/// <reference types="node" />
+/// <reference types="react" />
+
+declare module '*.svg' {
+  import * as React from 'react';
+  import { SxStyleProp } from 'theme-ui';
+
+  export const ReactComponent: React.FunctionComponent<
+    React.SVGProps<SVGSVGElement> & { sx?: SxStyleProp }
+  >;
+
+  export default ReactComponent;
+}
+
+declare module '@emotion/react' {
```

Comment:
Can you move this to `@types` directory? Please create `emotion.d.ts` file there and paste the content inside. Have you checked if the theme has the right type now when you're using it and not `any`?